src/Futhark/CLI/Unused.hs

```diff
@@ -0,0 +1,45 @@
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+
+{-# HLINT ignore "Use newtype instead of data" #-}
+
+-- | @futhark doc@
+module Futhark.CLI.Unused (main) where
+
+import Control.Monad.State
+import Data.Map.Strict qualified as M
+import Futhark.Compiler (dumpError, newFutharkConfig, readProgramFiles)
+import Futhark.Pipeline (Verbosity (..), runFutharkM)
+import Futhark.Util.Options
+import Language.Futhark
+import Language.Futhark.Unused
+import System.Exit
+import Data.List ( elemIndices )
+
+main :: String -> [String] -> IO ()
+main = mainWithOptions initialCheckConfig [] "files..." find
+  where
+    find [] _ = Nothing
+    find files _ = Just $ printUnused files
+
+printUnused :: [FilePath] -> IO ()
+printUnused files = do
+  res <- runFutharkM (readProgramFiles [] files) Verbose
+  case res of
+    Left err -> liftIO $ do
+      dumpError newFutharkConfig err
+      exitWith $ ExitFailure 2
+    Right (_, imp, _) -> do
+      putStrLn $
+        unlines $
```

Comment:
From what I remember, `unlines` will always add a final linebreak, and so will `putStrLn`.  So this will result in a blank line at the end.